biggus/__init__.py

```diff
@@ -824,6 +832,47 @@ def masked_array(self):
         return array
 
 
+def _use_indexed_array(keys):
+    return ((isinstance(keys, tuple) and any(key is None for key in keys)) or
+            keys is None)
+
+
+class IndexedArray(Array):
+    def __init__(self, array, keys):
+        self.array = array
+        if not isinstance(keys, tuple):
+            keys = (keys,)
+        self.keys = tuple(keys)
+
+    @property
+    def dtype(self):
+        return self.array.dtype
+
+    @property
+    def shape(self):
+        return _sliced_shape(self.array.shape, self.keys)
+
+    def __getitem__(self, keys):
+        raise RuntimeError('unsupported operation')
```

Comment:
Hmmm ... users might hit this brick wall pretty darn quick ...
```
>>> from biggus import NumpyArrayAdapter
>>> a = NumpyArrayAdapter(np.arange(20).reshape(4, 5))
>>> a
<NumpyArrayAdapter shape=(4, 5) dtype=dtype('int64')>
>>> b = a[:, None, :]
>>> b
<IndexedArray shape=(4, 1, 5) dtype=dtype('int64')>
>>> b[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/h05/itwl/projects/git/biggus/biggus/__init__.py", line 856, in __getitem__
    raise RuntimeError('unsupported operation')
RuntimeError: unsupported operation
```
Thoughts ... ?